lib/dom.js

```diff
@@ -1194,7 +1198,7 @@ declare class HTMLElement extends Element {
   className: string;
   contentEditable: string;
   contextMenu: ?HTMLMenuElement;
-  dataset: {[key:string]: string};
+  dataset: DOMStringMap;
```

Comment:
Isn't the dataset optional?
Or at least the keys are optional I think?

Reply:
The object type `{[key:string]: string}` has only an indexer defined. An indexer just means that **if** there are keys aside from the ones already defined, the keys and values must match those of the indexer. So all of the following inhabit the type `{[key:string]: string}`: `{}`, `{ foo: 'bar' }`, etc. Where as this would not inhabit the type: `{ 1: 'bar' }`, or `{ foo: 1 }`.

Reply:
Yeah I figured that out already, but thanks for making it clear to me once again 🙂